### PR TITLE
Add new setting for client validation

### PR DESCRIPTION
**Pull request by hany**
Needed for https://github.com/sensu/sensu/pull/1125.

### CODE REVIEW

**Comment by portertech** [comment]
:+1: Thank you.

File: lib/sensu/settings/validators/client.rb [hunk 1]
```diff
@@ -110,6 +110,8 @@ def validate_client(client)
               invalid(client, "client name cannot contain spaces or special characters")
             must_be_a_string(client[:address]) ||
               invalid(client, "client address must be a string")
+            must_be_a_string_if_set(client[:client_key]) ||
```
Comment (portertech) [inline lib/sensu/settings/validators/client.rb]: shouldn't this attribute just be `key`? Following the normal definition attribute naming convention? Using scope.

Comment (hany) [inline lib/sensu/settings/validators/client.rb]: Hey @portertech. I opted for `client_key` as `key` seemed a bit too ambiguous. Naming things is always hard. `client_key` tells you it's a client-side value as opposed to some authentication key that the server would require. Naming things is always hard...

**Comment by portertech** [comment]
Going to close this, in favour of https://github.com/sensu/sensu-settings/pull/23 :+1:

**Pull request closed without merging**